### PR TITLE
Migrating ui internals to new resource implementation 3/3 (ref-point migration 8/10)

### PR DESCRIPTION
**Pull request by tobous**
Migrates parts of the internal UI logic of Saros/E to use the new resource model introduced in #1023. Outdated javadoc and method names that are not directly related to the changed logic will be updated in a separate clean-up PR at the end.

This is the third and final part of the internal UI migration.

### Reviewing this PR

I would suggest reviewing this PR commit by commit.

### Commits

<details><summary><b>[INTERNAL][E] Simplify SelectionRetrieverFactory</b></summary>
<br>

Removes the "smart" functionality converting from the selected resource
to its project as it is no longer needed. With the new resource model,
we are actually interested in the specific resources the user selected.

Removes AbstractSelectionConvertingRetriever as it is no longer being
used.

</details>

<details><summary><b>[UI][E] Adjust UI integration to show sharing pop up on all resources</b></summary>
<br>

Adjusts the plugin.xml configuration of the project view context menu
integration to show the option to share the selected resources when any
kind of resource is selected. This was done as the new sharing logic
allows to show any container as a base resource and not just projects.

</details>

<b>[UI][E] Remove last references to sharing projects shown to the user</b>

### CODE REVIEW

**Comment by saros-infrastructure** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity decreasing per file
==============================
+ eclipse/src/saros/ui/util/selection/retriever/SelectionRetrieverFactory.java  -2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/Saros/saros/pullRequest?prid=5736999&bid=18766411)

**Comment by tobous** [comment]
Rebased onto current base branch without any interaction.

**Comment by saros-infrastructure** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity decreasing per file
==============================
+ eclipse/src/saros/ui/util/selection/retriever/SelectionRetrieverFactory.java  -2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/Saros/saros/pullRequest?prid=5736999&bid=18766411)

**Review by srossbach — DISMISSED** [review]

**Comment by tobous** [comment]
Rebased onto current base branch without any interaction.

**Comment by saros-infrastructure** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity decreasing per file
==============================
+ eclipse/src/saros/ui/util/selection/retriever/SelectionRetrieverFactory.java  -2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/Saros/saros/pullRequest?prid=5736999&bid=18766411)

File: eclipse/plugin.xml [hunk 1]
```diff
@@ -514,7 +514,7 @@
       <command
             categoryId="org.eclipse.ui.category.project"
             id="saros.ui.commands.ShareResource"
-            name="Share Project(s)...">
+            name="Share Resource(s)...">
```
Comment (srossbach) [inline eclipse/plugin.xml:517]: Resource is an internal Eclipse specific API object.  I do not know if this is the best name we should use.

Comment (tobous) [reply eclipse/plugin.xml:517]: What else should we call it?

I was first going to call it "Share Reference Points" (and was going to name the negotiation "reference point negotiation", but Kelvin pointed out that "sharing resources" is an easy concept for the user to understand, so I decided to use it.

Comment (srossbach) [reply eclipse/plugin.xml:517]: Are there any plans for testing the usability changes of your patches ? If this is the case you can stick with Resource for now and wait for any issues that might arise with the new sharing logic.

Comment (tobous) [reply eclipse/plugin.xml:517]: > Are there any plans for testing the usability changes of your patches ? If this is the case you can stick with Resource for now and wait for any issues that might arise with the new sharing logic.

I did not have any further usability test planned, no. But this should not be too hard to change later on.

As already mentioned, I would like to mainly concentrate on writing the actual thesis document for now (and getting the existing code reviewed). I already expanded the scope of the thesis, but I have to draw the line at some point. (I am still working with a deadline.)

**Comment by tobous** [comment]
Rebased onto current base branch without any interaction.

**Comment by saros-infrastructure** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity decreasing per file
==============================
+ eclipse/src/saros/ui/util/selection/retriever/SelectionRetrieverFactory.java  -2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/Saros/saros/pullRequest?prid=5736999&bid=18766411)

**Comment by tobous** [comment]
Rebased onto current base branch without any interactions.

**Comment by saros-infrastructure** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity decreasing per file
==============================
+ eclipse/src/saros/ui/util/selection/retriever/SelectionRetrieverFactory.java  -2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/Saros/saros/pullRequest?prid=5736999&bid=18766411)

**Comment by tobous** [comment]
Rebased onto current base branch without any interaction.

**Comment by saros-infrastructure** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity decreasing per file
==============================
+ eclipse/src/saros/ui/util/selection/retriever/SelectionRetrieverFactory.java  -2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/Saros/saros/pullRequest?prid=5736999&bid=18766411)

**Comment by tobous** [comment]
Rebased onto current master without any interaction.

**Comment by saros-infrastructure** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity decreasing per file
==============================
+ eclipse/src/saros/ui/util/selection/retriever/SelectionRetrieverFactory.java  -2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/Saros/saros/pullRequest?prid=5736999&bid=18766411)

**Pull request merged**